src/interactive.js

```diff
@@ -1,75 +1,183 @@
+      const todos = JSON.parse(localStorage.getItem('todos'));
+      check.forEach((check, index) => {
+        check.addEventListener('change', (e) => {
+          e.preventDefault();
+          const divWrapper = check.parentElement;
+          if (check.checked) {
+            divWrapper.classList.add('checked');
+            check.checked = true;
+            todos[index].completed = true;
+            localStorage.setItem('todos', JSON.stringify(todos));
+          } else {
+            divWrapper.classList.remove('checked');
+            todos[index].completed = false;
+            localStorage.setItem('todos', JSON.stringify(todos));
+          }
+        });
+        window.addEventListener('load', (e) => {
+          e.preventDefault();
+          const divWrapper = check.parentElement;
+          if (check.checked) {
+            divWrapper.classList.add('checked');
+            check.checked = true;
+            todos[index].completed = true;
+            localStorage.setItem('todos', JSON.stringify(todos));
+          } else {
+            divWrapper.classList.remove('checked');
+            todos[index].completed = false;
+            localStorage.setItem('todos', JSON.stringify(todos));
+          }
+        });
+      });
     }
+  }
 
-    const check = document.querySelectorAll('.check-input');
-    const todos = JSON.parse(localStorage.getItem('todos'));
-    check.forEach((check, index) => {
-      check.addEventListener('change', (e) => {
+  editItem() {
+    const edit = document.querySelectorAll('.edit');
+    const todoList = JSON.parse(localStorage.getItem('todos'));
+    edit.forEach((v, i) => {
+      v.addEventListener('click', (e) => {
         e.preventDefault();
-        const divWrapper = check.parentElement;
-        if (check.checked) {
-          divWrapper.classList.add('checked');
-          check.checked = true;
-          todos[index].completed = true;
-          localStorage.setItem('todos', JSON.stringify(todos));
-        } else {
-          divWrapper.classList.remove('checked');
-          todos[index].completed = false;
-          localStorage.setItem('todos', JSON.stringify(todos));
-        }
+        const aDiv = v.parentElement;
+        const listItem = aDiv.parentElement;
+        const p = listItem.children[1];
+
+        todoList[i].todoText = p.innerText;
+        localStorage.setItem('todos', JSON.stringify(todoList));
       });
+    });
+  }
+
+  delItem() {
+    const del = document.querySelectorAll('.del');
+    const todoList = JSON.parse(localStorage.getItem('todos'));
+    del.forEach((v, i) => {
+      setTimeout(() => {
+        const aDiv = v.parentElement;
+        const listItem = aDiv.parentElement.parentElement;
+        v.addEventListener('click', (e) => {
+          e.preventDefault();
+          listItem.remove();
+          todoList.splice(i, 1);
+          this.addItem(todoList);
+        });
+      }, 500);
+    });
+  }
+
+  clearAll() {
```

Comment:
- [ ] When i press the button that triggers this function, the application stop working, please check and solve this issue 😄